staging/src/k8s.io/kube-aggregator/pkg/controllers/status/remote/remote_available_controller.go

```diff
@@ -239,30 +240,38 @@ func (c *AvailableConditionController) sync(key string) error {
 			return err
 		}
 
-		endpoints, err := c.endpointsLister.Endpoints(apiService.Spec.Service.Namespace).Get(apiService.Spec.Service.Name)
-		if apierrors.IsNotFound(err) {
-			availableCondition.Status = apiregistrationv1.ConditionFalse
-			availableCondition.Reason = "EndpointsNotFound"
-			availableCondition.Message = fmt.Sprintf("cannot find endpoints for service/%s in %q", apiService.Spec.Service.Name, apiService.Spec.Service.Namespace)
-			apiregistrationv1apihelper.SetAPIServiceCondition(apiService, availableCondition)
-			_, err := c.updateAPIServiceStatus(originalAPIService, apiService)
-			return err
-		} else if err != nil {
+		endpointSliceSelector := labels.SelectorFromSet(labels.Set{discoveryv1.LabelServiceName: apiService.Spec.Service.Name})
+		endpointSlices, err := c.endpointSliceLister.EndpointSlices(apiService.Spec.Service.Namespace).List(endpointSliceSelector)
```

Comment:
same comment about wondering if this should make use of an indexer, the POC commit I linked to also fixes up this call site